app/jinja_filters.py

```diff
@@ -409,6 +409,9 @@ def __init__(self, text, other=None):
         if other or other == 0:
             self.other = other
 
+    def __eq__(self, other):
+        return self.text == other.text
```

Comment:
It is a good practice to check the instance is of same type. Also, you could compare the `__dict__` dunder methods since these are simple object, and it would keep them up to date if new attributes got added. In your definition, two `SummaryRowItemValue` objects with same text would return True even if the value of `self.other` was different.
```py
    def __eq__(self, other):
        return (
            isinstance(other, SummaryRowItemValue) and self.__dict__ == other.__dict__
        )
```
Applies to the other objects also.

Reply:
Are these `__eq__` methods needed? The `map_summary_item_config_processor` method casts the return from `map_summary_item_config` as a [dict](https://github.com/ONSdigital/eq-questionnaire-runner/blob/master/app/jinja_filters.py#L572), could the same be done in the test?

Reply:
> Are these `__eq__` methods needed? The `map_summary_item_config_processor` method casts the return from `map_summary_item_config` as a [dict](https://github.com/ONSdigital/eq-questionnaire-runner/blob/master/app/jinja_filters.py#L572), could the same be done in the test?

I tested this and whenever `SummaryRowItem` and `SummaryRow` extends some class/classes inside it, it fails the unit test when I cast return as `dict`, so `dict(expected) == dict(result)` always fails. I get type errors.

Reply:
> It is a good practice to check the instance is of same type. Also, you could compare the `__dict__` dunder methods since these are simple object, and it would keep them up to date if new attributes got added. In your definition, two `SummaryRowItemValue` objects with same text would return True even if the value of `self.other` was different.
> 
> ```python
>     def __eq__(self, other):
>         return (
>             isinstance(other, SummaryRowItemValue) and self.__dict__ == other.__dict__
>         )
> ```
> 
> Applies to the other objects also.

I could only compare using `isinstance()` when I try `__dict__` on classes that extend some other classes it fails..

Reply:
> It is a good practice to check the instance is of same type. Also, you could compare the `__dict__` dunder methods since these are simple object, and it would keep them up to date if new attributes got added. In your definition, two `SummaryRowItemValue` objects with same text would return True even if the value of `self.other` was different.
> 
> ```python
>     def __eq__(self, other):
>         return (
>             isinstance(other, SummaryRowItemValue) and self.__dict__ == other.__dict__
>         )
> ```
> 
> Applies to the other objects also.

I added this everywhere except `SummaryRowItem` which fails tests if I add `self.__dict__ == other.__dict__` to its `__eq__` method.

Reply:
@adammahmood I wasn't aware you can call `map_summary_item_config_processor` directly and get a dict back of all objects, thought that was a Flask specific thing that did the conversion to dict. If it can, then we don't need `__eq__`.